src/env.types.ts

```diff
@@ -0,0 +1,47 @@
+export type Env = GitLabCIPredefinedVariables &
+  MergeRequestVariables & {
+    GITLAB_HOST: string
+
+    GITLAB_TOKEN: string
+    GITLAB_TOKEN_TYPE: LooseString<'job' | 'oauth'>
+    GITLAB_CI_USER_NAME: string | undefined
+    GITLAB_CI_USER_EMAIL: string
+    GITLAB_COMMENT_TYPE: LooseString<'discussion' | 'note'>
+    DEBUG_GITLAB_CREDENTIAL: LooseString<'1' | 'true'>
+
+    HOME: string
+    NPM_TOKEN: string | undefined
+  }
+
+type MergeRequestVariables =
+  | {
+      // this is used to be checked if the current job is a merge request job
+      CI_MERGE_REQUEST_SOURCE_BRANCH_NAME: undefined
+    }
+  | {
+      CI_MERGE_REQUEST_IID: number
```

Comment:
How can it be cast to `number` automatically?

Reply:
Oh, this is my fault, I forgot to include it in the `prepareEnv` function.